File: src/frontend/src/models/RevisionDto.ts
export default interface RevisionDto {
  id: number
  studentTesteId: number
  studentName: string
  testeTitle: string
  currentGrade: number
  justification: string
  status: 'PENDING' | 'TA_GRADED' | 'APPROVED' | 'REJECTED'
  requestDate: string
  reviewedBy?: string
  reviewDate?: string
  newGrade?: number
}

export interface CreateRevisionDto {
  studentTesteId: number
  justification: string
}

File: src/backend/src/main/java/pt/ulisboa/tecnico/rnl/dei/dms/assessments/testes/revisions/dto/RevisionDto.java
package pt.ulisboa.tecnico.rnl.dei.dms.assessments.testes.revisions.dto;

import pt.ulisboa.tecnico.rnl.dei.dms.assessments.testes.revisions.domain.Revision;

import java.time.LocalDateTime;

public record RevisionDto(
    Long id,
    Long studentTesteId,
    String studentName,
    String testeTitle,
    Float currentGrade,
    String justification,
    String status,
    LocalDateTime requestDate,
    String reviewedBy,
    LocalDateTime reviewDate,
    Float newGrade
) {
    public RevisionDto(Revision revision) {
        this(
            revision.getId(),
            revision.getStudentTeste().getId(),
            revision.getStudentTeste().getStudent().getName(),
            revision.getStudentTeste().getTeste().getTitle(),
            revision.getStudentTeste().getGrade(),
            revision.getJustification(),
            revision.getStatus().toString(),
            revision.getRequestDate(),
            revision.getReviewedBy() != null ? revision.getReviewedBy().getName() : null,
            revision.getReviewDate(),
            revision.getNewGrade()
        );
    }
}